synapse/federation/transport/server.py

```diff
@@ -544,15 +544,17 @@ async def on_PUT(self, origin, content, query, room_id, event_id):
 
 
 class FederationMakeKnockServlet(BaseFederationServlet):
-    PATH = "/make_knock/(?P<room_id>[^/]*)/(?P<user_id>[^/]*)"
+    PATH = "/make_xyz.amorgan.knock/(?P<room_id>[^/]*)/(?P<user_id>[^/]*)"
 
     async def on_GET(self, origin, content, query, room_id, user_id):
         content = await self.handler.on_make_knock_request(origin, room_id, user_id)
         return 200, content
 
 
 class FederationV2SendKnockServlet(BaseFederationServlet):
-    PATH = "/send_knock/(?P<room_id>[^/]*)/(?P<event_id>[^/]*)"
+    PATH = "/send_xyz.amorgan.knock/(?P<room_id>[^/]*)/(?P<event_id>[^/]*)"
+
+    PREFIX = FEDERATION_V2_PREFIX
```

Comment:
Shouldn't this be using an unstable prefix with the "normal" endpoint? e.g. `/xyz.amorgan.knock/send_knock` or something like that?

Reply:
Oh yes huh, and also it needs to **not** be available under `r0` on the CS API. Good catch!